faker/providers/internet/__init__.py

```diff
@@ -99,15 +102,39 @@ def url(self):
         pattern = self.random_element(self.url_formats)
         return self.generator.parse(pattern)
 
-    def ipv4(self):
+    def ipv4(self, network=False):
+        """ Produce a random IPv4 address or network with a valid CIDR.
+
+        IP code generation inspired by ipaddress module: https://github.com/phihag/ipaddress
```

Comment:
Done in commit: https://github.com/kdeldycke/faker/commit/d16ea3c7a76d530ec64c9ab0d3533346618be6df